### PR TITLE
Extend Azure template for all use cases

### PR DESCRIPTION
**Pull request by pho-enix**

### CODE REVIEW

**Comment by pho-enix** [comment]
Please keep in mind: That's all public ...

File: mainTemplate.json [hunk 1]
```diff
@@ -203,6 +205,27 @@
             }
         },
         {
+            "type": "Microsoft.Network/routeTables",
+            "name": "[variables('routeTableName')]",
+            "apiVersion": "[variables('apiVersion').network]",
+            "location": "[resourceGroup().location]",
+            "dependsOn": [
+                "[concat('Microsoft.Network/networkInterfaces/', variables('nicName2'))]"
```
Comment (threez) [inline mainTemplate.json]: please use nicWAN and nicLAN as nic names

File: createUiDefinition.json [hunk 2]
```diff
@@ -183,10 +183,13 @@
             "wanPrefix": "[steps('instanceDetails').vnetSpec.subnets.subnet1.addressPrefix]",
             "lanName": "[steps('instanceDetails').vnetSpec.subnets.subnet2.name]",
             "lanPrefix": "[steps('instanceDetails').vnetSpec.subnets.subnet2.addressPrefix]",
-            "domainNameLabel": "[steps('instanceDetails').domainSpec.domainNameLabel]",
             "storageName": "[steps('instanceDetails').storageSpec.name]",
             "storageType": "[steps('instanceDetails').storageSpec.type]",
-            "imageSku": "[steps('instanceDetails').license]"
+            "imageSku": "[steps('instanceDetails').license]",
```
Comment (threez) [inline createUiDefinition.json:190]: please imageSKU

File: createUiDefinition.json [hunk 3]
```diff
@@ -183,10 +183,13 @@
...
+            "pipDNS": "[steps('instanceDetails').domainSpec.domainNameLabel]",
```
Comment (threez) [inline createUiDefinition.json]: please use publicIPName ... pip is a bit confusing

File: mainTemplate.json [hunk 4]
```diff
@@ -75,18 +69,41 @@
             "metadata": {
                 "description": "License type to use"
             }
+        },
+        "pipDNS": {
+            "type": "string",
+            "metadata": {
+                "description": "Unique DNS name prefix (e.g. sfostest1)"
```
Comment (threez) [inline mainTemplate.json]: remove the test1 part

File: LICENSE.txt [hunk 5]
```diff
@@ -1,23 +1,26 @@
-Copyright © 2016 Sophos Limited. All rights reserved. 
+The MIT License (MIT)
 
-Sophos is a registered trademark of Sophos Limited and Sophos Group. All other 
-product and company names mentioned are trademarks or registered trademarks of 
+Copyright © 2016 Sophos Limited. All rights reserved.
+Copyright © 2015 Microsoft Azure
```
Comment (jpwenzel) [inline LICENSE.txt]: We should use separate license files for our own and Microsoft's property.

**Comment by aheck** [comment]
Looks good, except for the thing with the license file

**Pull request merged**